### PR TITLE
 #644: Fix undefined callback functions inspection (-optional_callback() attribute supported)

### PR DESCRIPTION
**Pull request by telezhnaya**

### CODE REVIEW

File: src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java [hunk 1]
```diff
@@ -1718,6 +1715,27 @@ public static boolean isExported(@NotNull ErlangFunction o) {
...
+
+    for (ErlangCallbackFunction callback : file.getOptionalCallbacks()) {
+      ErlangAtom functionNameAtom = callback.getQAtom().getAtom();
+      String functionName = functionNameAtom != null ? getName(functionNameAtom) : null;
+      if (specName.equals(functionName) && specArity == getArity(callback.getInteger())) {
+        isOptional = true;
+        break;
+      }
+    }
+    return isOptional;
```
Comment (deadok22) [inline src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java]: There is no reason to have this ```isOptional``` variable - it only complicates reading.

File: src/org/intellij/erlang/psi/impl/ErlangFileImpl.java [hunk 2]
```diff
@@ -712,6 +712,32 @@ public boolean process(PsiElement psiElement) {
...
+    final List<ErlangCallbackFunction> result = ContainerUtil.newArrayList();
+    for (ErlangAttribute attribute : getAttributes()) {
+      ErlangOptionalCallbacks callbacks = attribute.getOptionalCallbacks();
+      ErlangOptionalCallbackFunctions functions = callbacks != null ? callbacks.getOptionalCallbackFunctions() : null;
+      if (functions == null) continue;
+      for (ErlangCallbackFunction function : functions.getCallbackFunctionList()) {
+        result.add(function);
+      }
+    }
+    return result;
```
Comment (deadok22) [inline src/org/intellij/erlang/psi/impl/ErlangFileImpl.java]: You compute these callback functions on every ```getOptionalCallbacks``` call. Take a look at other methods in this class - they all make use of caching - you should, too.

**Comment by telezhnaya** [comment]
Check it please.

File: src/org/intellij/erlang/psi/impl/ErlangFileImpl.java [hunk 3]
```diff
@@ -712,6 +712,38 @@ public boolean process(PsiElement psiElement) {
...
+        ErlangOptionalCallbackFunctions functions = callbacks != null ? callbacks.getOptionalCallbackFunctions() : null;
+        if (functions != null) {
+          for (ErlangCallbackFunction function : functions.getCallbackFunctionList()) {
+            result.add(function);
+          }
+        }
+        return true;
+      }
+    });
+    return result;
```
Comment (deadok22) [inline src/org/intellij/erlang/psi/impl/ErlangFileImpl.java]: I asked you to employ caching for this value in previous review.

**Comment by telezhnaya** [comment]
Check it please.

**Comment by deadok22** [comment]
Your tests are failing...

**Comment by telezhnaya** [comment]
I am really sorry. Check it again please. All tests passed.

File: src/org/intellij/erlang/psi/impl/ErlangFileImpl.java [hunk 4]
```diff
@@ -712,6 +719,27 @@ public boolean process(PsiElement psiElement) {
...
+  private Collection<ErlangCallbackFunction> calcOptionalCallbacks() {
+    ErlangFileStub stub = getStub();
+    if (stub != null) {
+      return getChildrenByType(stub, ErlangTypes.ERL_CALLBACK_FUNCTION, new ArrayFactory<ErlangCallbackFunction>() {
+        @NotNull
+        @Override
+        public ErlangCallbackFunction[] create(int count) {
+          return new ErlangCallbackFunction[count];
+        }
+      });
```
Comment (deadok22) [inline src/org/intellij/erlang/psi/impl/ErlangFileImpl.java]: Please, preserve code style.  All other ```calc***``` methods here reference corresponding stub type's ARRAY_FACTORY field. Use the one in org.intellij.erlang.stubs.types.ErlangCallbackStubElementType. Not only you'll reuse existing object, but you'll also improve your code readability.

**Comment by telezhnaya** [comment]
Check it please.

**Comment by deadok22** [comment]
Please, annotate method parameters and return values with nullability annotations.

**Comment by telezhnaya** [comment]
Annotations added.

File: src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java [hunk 5]
```diff
@@ -48,6 +48,7 @@ protected void checkFile(@NotNull ErlangFile file, @NotNull ProblemsHolder probl
       List<ErlangCallbackSpec> undefinedCallbacks = ContainerUtil.newArrayList();
       Map<String, ErlangCallbackSpec> callbackMap = behaviourModule.getCallbackMap();
       for (ErlangCallbackSpec spec : callbackMap.values()) {
+        if (spec.isOptional()) continue;
```
Comment (deadok22) [inline src/org/intellij/erlang/inspection/ErlangUndefinedCallbackFunctionInspection.java]: Optional callback attribute should be taken into account for release ```18.0``` and above.

**Comment by telezhnaya** [comment]
Check it please.

**Comment by deadok22** [comment]
You've updated stubs without pushing stub version up. Here is what it looks like from users' point of view:

```
java.lang.RuntimeException: java.io.EOFException
	at com.intellij.psi.stubs.SerializationManagerImpl.deserialize(SerializationManagerImpl.java:178)
	at com.intellij.psi.stubs.SerializedStubTree.getStub(SerializedStubTree.java:69)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readFromVFile(StubTreeLoaderImpl.java:109)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readOrBuild(StubTreeLoaderImpl.java:47)
	at com.intellij.psi.impl.source.PsiFileImpl.getStubTree(PsiFileImpl.java:681)
	at com.intellij.psi.impl.source.PsiFileImpl.getStub(PsiFileImpl.java:661)
	at org.intellij.erlang.psi.impl.ErlangFileImpl.getStub(ErlangFileImpl.java:302)
	at org.intellij.erlang.psi.impl.ErlangFileImpl.calcFunctions(ErlangFileImpl.java:792)
	at org.intellij.erlang.psi.impl.ErlangFileImpl.access$100(ErlangFileImpl.java:58)
	at org.intellij.erlang.psi.impl.ErlangFileImpl$2.compute(ErlangFileImpl.java:113)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:200)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at org.intellij.erlang.psi.impl.ErlangFileImpl.getFunctions(ErlangFileImpl.java:503)
	at org.intellij.erlang.editor.ErlangFoldingBuilder.buildFoldRegions(ErlangFoldingBuilder.java:46)
	at com.intellij.lang.folding.LanguageFolding.buildFoldingDescriptors(LanguageFolding.java:83)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:265)
	at com.intellij.codeInsight.folding.impl.FoldingUpdate.getFoldingsFor(FoldingUpdate.java:250)
	at com.intellij.codeInsight.folding.impl.CodeFoldingManagerImpl.buildInitialFoldings(CodeFoldingManagerImpl.java:232)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorProvider.createEditorAsync(PsiAwareTextEditorProvider.java:63)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$11.compute(FileEditorManagerImpl.java:823)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$11.compute(FileEditorManagerImpl.java:816)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:888)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl4(FileEditorManagerImpl.java:816)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.processFiles(EditorsSplitters.java:859)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.processFiles(EditorsSplitters.java:829)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$ConfigTreeReader.process(EditorsSplitters.java:820)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters.openFiles(EditorsSplitters.java:244)
	at com.intellij.openapi.fileEditor.impl.OpenFilesActivity.runActivity(OpenFilesActivity.java:35)
	at com.intellij.ide.startup.impl.StartupManagerImpl$2.run(StartupManagerImpl.java:156)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:339)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runPostStartupActivitiesFromExtensions(StartupManagerImpl.java:161)
	at com.intellij.openapi.project.impl.ProjectManagerImpl$8.run(ProjectManagerImpl.java:490)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:237)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
	at com.intellij.openapi.progress.impl.CoreProgressManager$8.run(CoreProgressManager.java:367)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at com.intellij.openapi.application.impl.ApplicationImpl$10$1.run(ApplicationImpl.java:617)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)
Caused by: java.io.EOFException
	at java.io.DataInputStream.readUnsignedByte(DataInputStream.java:290)
	at com.intellij.util.io.DataInputOutputUtil.readINT(DataInputOutputUtil.java:43)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:161)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:163)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:140)
	at com.intellij.psi.stubs.SerializationManagerImpl.deserialize(SerializationManagerImpl.java:173)
	... 48 more
[   7905]  ERROR - ext.PsiAwareTextEditorProvider - IntelliJ IDEA (Community Edition) IC-141.1531.2  Build #IC-141.1531.2 
[   7905]  ERROR - ext.PsiAwareTextEditorProvider - JDK: 1.8.0_45 
[   7905]  ERROR - ext.PsiAwareTextEditorProvider - VM: Java HotSpot(TM) 64-Bit Server VM 
[   7905]  ERROR - ext.PsiAwareTextEditorProvider - Vendor: Oracle Corporation 
[   7905]  ERROR - ext.PsiAwareTextEditorProvider - OS: Linux 
[   7905]  ERROR - ext.PsiAwareTextEditorProvider - Last Action:  
AttributeStubFileStubSerializerjava.MODIFIER_LISTjava.ANNOTATIONjava.ANN; parent stub class=org.intellij.erlang.stubs.ErlangCallbackSpecStub 
com.intellij.psi.stubs.SerializerNotFoundException: No serializer registered for stub: ID=6, externalId:ileElementStubSerializer
AttributeStubFileStubSerializerjava.MODIFIER_LISTjava.ANNOTATIONjava.ANN; parent stub class=org.intellij.erlang.stubs.ErlangCallbackSpecStub
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:157)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:163)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:163)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:140)
	at com.intellij.psi.stubs.SerializationManagerImpl.deserialize(SerializationManagerImpl.java:173)
	at com.intellij.psi.stubs.SerializedStubTree.getStub(SerializedStubTree.java:69)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readFromVFile(StubTreeLoaderImpl.java:109)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readOrBuild(StubTreeLoaderImpl.java:47)
	at com.intellij.psi.impl.source.PsiFileImpl.getStubTree(PsiFileImpl.java:681)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:58)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:39)
	at com.intellij.psi.stubs.StubIndexImpl$2.perform(StubIndexImpl.java:274)
	at com.intellij.psi.stubs.StubIndexImpl$2.perform(StubIndexImpl.java:265)
	at com.intellij.util.indexing.ValueContainer.forEach(ValueContainer.java:81)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:265)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:238)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:76)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:95)
	at com.intellij.psi.stubs.StubIndexImpl.get(StubIndexImpl.java:227)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:144)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:134)
	at org.intellij.erlang.stubs.index.ErlangBehaviourModuleIndex.getModules(ErlangBehaviourModuleIndex.java:47)
	at org.intellij.erlang.completion.ErlangCompletionContributor.suggestBehaviours(ErlangCompletionContributor.java:439)
	at org.intellij.erlang.completion.ErlangCompletionContributor.access$500(ErlangCompletionContributor.java:72)
	at org.intellij.erlang.completion.ErlangCompletionContributor$1.addCompletions(ErlangCompletionContributor.java:203)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:135)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:128)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$2.addCompletions(LiveTemplateCompletionContributor.java:90)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:112)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:796)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.access$1000(CompletionProgressIndicator.java:90)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator$1CalculateItems.run(CompletionProgressIndicator.java:778)
	at com.intellij.codeInsight.completion.AsyncCompletion$1$1$1.run(CompletionThreading.java:93)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:872)
	at com.intellij.codeInsight.completion.AsyncCompletion$1$1.run(CompletionThreading.java:88)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at com.intellij.codeInsight.completion.AsyncCompletion$1.run(CompletionThreading.java:84)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)
[  59687]  ERROR - j.psi.stubs.StubTreeLoaderImpl - IntelliJ IDEA (Community Edition) IC-141.1531.2  Build #IC-141.1531.2 
[  59687]  ERROR - j.psi.stubs.StubTreeLoaderImpl - JDK: 1.8.0_45 
[  59687]  ERROR - j.psi.stubs.StubTreeLoaderImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[  59687]  ERROR - j.psi.stubs.StubTreeLoaderImpl - Vendor: Oracle Corporation 
[  59687]  ERROR - j.psi.stubs.StubTreeLoaderImpl - OS: Linux 
[  59688]  ERROR - j.psi.stubs.StubTreeLoaderImpl - Last Action: CodeCompletion 
[  59944]  ERROR - on.CompletionProgressIndicator - org.intellij.erlang.stubs.ErlangCallbackSpecStub cannot be cast to com.intellij.util.xml.stubs.ElementStub 
java.lang.ClassCastException: org.intellij.erlang.stubs.ErlangCallbackSpecStub cannot be cast to com.intellij.util.xml.stubs.ElementStub
	at com.intellij.util.xml.stubs.ElementStubSerializer.deserialize(ElementStubSerializer.java:33)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:160)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:163)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:163)
	at com.intellij.psi.stubs.StubSerializationHelper.deserialize(StubSerializationHelper.java:140)
	at com.intellij.psi.stubs.SerializationManagerImpl.deserialize(SerializationManagerImpl.java:173)
	at com.intellij.psi.stubs.SerializedStubTree.getStub(SerializedStubTree.java:69)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readFromVFile(StubTreeLoaderImpl.java:109)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readOrBuild(StubTreeLoaderImpl.java:47)
	at com.intellij.psi.impl.source.PsiFileImpl.getStubTree(PsiFileImpl.java:681)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:58)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:39)
	at com.intellij.psi.stubs.StubIndexImpl$2.perform(StubIndexImpl.java:274)
	at com.intellij.psi.stubs.StubIndexImpl$2.perform(StubIndexImpl.java:265)
	at com.intellij.util.indexing.ValueContainer.forEach(ValueContainer.java:81)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:265)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:238)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:76)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:95)
	at com.intellij.psi.stubs.StubIndexImpl.get(StubIndexImpl.java:227)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:144)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:134)
	at org.intellij.erlang.stubs.index.ErlangBehaviourModuleIndex.getModules(ErlangBehaviourModuleIndex.java:47)
	at org.intellij.erlang.completion.ErlangCompletionContributor.suggestBehaviours(ErlangCompletionContributor.java:439)
	at org.intellij.erlang.completion.ErlangCompletionContributor.access$500(ErlangCompletionContributor.java:72)
	at org.intellij.erlang.completion.ErlangCompletionContributor$1.addCompletions(ErlangCompletionContributor.java:203)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:135)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:128)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$2.addCompletions(LiveTemplateCompletionContributor.java:90)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:112)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:796)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.access$1000(CompletionProgressIndicator.java:90)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator$1CalculateItems.run(CompletionProgressIndicator.java:778)
	at com.intellij.codeInsight.completion.AsyncCompletion$1$1$1.run(CompletionThreading.java:93)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:872)
	at com.intellij.codeInsight.completion.AsyncCompletion$1$1.run(CompletionThreading.java:88)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:152)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:137)
	at com.intellij.codeInsight.completion.AsyncCompletion$1.run(CompletionThreading.java:84)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)
[  59945]  ERROR - on.CompletionProgressIndicator - IntelliJ IDEA (Community Edition) IC-141.1531.2  Build #IC-141.1531.2 
[  59945]  ERROR - on.CompletionProgressIndicator - JDK: 1.8.0_45 
[  59945]  ERROR - on.CompletionProgressIndicator - VM: Java HotSpot(TM) 64-Bit Server VM 
[  59945]  ERROR - on.CompletionProgressIndicator - Vendor: Oracle Corporation 
[  59945]  ERROR - on.CompletionProgressIndicator - OS: Linux 
[  59946]  ERROR - on.CompletionProgressIndicator - Last Action: CodeCompletion 
```

**Comment by telezhnaya** [comment]
Check it please.

**Pull request merged**